pkg/generators/openapi_test.go

```diff
@@ -1693,3 +1694,80 @@ Extensions: spec.Extensions{
 
 `, funcBuffer.String())
 }
+
+func TestSymbolReference(t *testing.T) {
+	callErr, funcErr, assert, _, funcBuffer, imports := testOpenAPITypeWriter(t, `
+package foo
+
+const MyConst string = "Hello"
+
+// +k8s:openapi-gen=true
+type Blah struct {
+	// +default="A Default Value"
+	// +optional
+	Value *string
+
+	// +default=ref(MyConst)
+	// +optional
+	OtherValue *string
+
+	// +default=ref(base/foo.MyConst)
+	// +optional
+	FullyQualifiedLocalPackage *string
+
+	// +default=ref(k8s.io/api/v1.TerminationPathDefault)
+	// +optional
+	FullyQualifiedExternalPackage *string
+}
+	`)
+	assert.NoError(funcErr)
+	assert.NoError(callErr)
+	assert.Equal(imports, []string{`v1 "k8s.io/api/v1"`})
+
+	if formatted, err := format.Source(funcBuffer.Bytes()); err != nil {
+		t.Fatal(err)
+	} else {
+		assert.Equal(string(formatted), `func schema_base_foo_Blah(ref common.ReferenceCallback) common.OpenAPIDefinition {
+	return common.OpenAPIDefinition{
+		Schema: spec.Schema{
+			SchemaProps: spec.SchemaProps{
+				Type: []string{"object"},
+				Properties: map[string]spec.Schema{
+					"Value": {
+						SchemaProps: spec.SchemaProps{
+							Default: "A Default Value",
+							Type:    []string{"string"},
+							Format:  "",
+						},
+					},
+					"OtherValue": {
+						SchemaProps: spec.SchemaProps{
+							Default: MyConst,
```

Comment:
is the variable actually copied into the openapi file that has the schema?

Reply:
you might have to fully qualify the local package when you do that since I don't think the generate openapi is going in the same package as the original types.

Reply:
Just tested on k8s upstream, it has the correct package since the ImportNamer takes in the path to the output package - it knows the output is different from the package owning the comment.